### PR TITLE
Monitor endpoint changes as well.

### PR DESCRIPTION
**Pull request by AbkDj**
Initial Pull request. Added changes to monitor ingress and endpoints and configure NS. Also added python script to perform NS VXLAN configuration and ETCD NS MAC address configuration.

### CODE REVIEW

File: ingress/controllers/citrix-netscaler/netscaler/netscaler_resource.go [hunk 1]
```diff
@@ -183,11 +183,11 @@ func listBoundResources(resourceName string, resourceType string, boundResourceT
 
 	client := &http.Client{}
 	resp, err := client.Do(req)
-	defer resp.Body.Close()
 	if err != nil {
 		log.Fatal(err)
 		return []byte{}, err
 	} else {
+		defer resp.Body.Close()
```
Comment (chiradeep) [inline ingress/controllers/citrix-netscaler/netscaler/netscaler_resource.go:190]: why did you move it here?

Comment (AbkDj) [inline ingress/controllers/citrix-netscaler/netscaler/netscaler_resource.go:190]: In case of an error "resp" is NULL and using it in the next line causes the controller to crash. The defer is added if there is no error. Otherwise we return.
